controllers/AchievementsController.js

```diff
@@ -0,0 +1,11 @@
+const User = require("../models/user");
+
+exports.getUserAchievements = (req, res, next) => {
+    if (!req.params.username) return res.status(400).json({success: false});
+    const name = req.params.username;
+    User.findOne({name}).then(user => {
```

Comment:
Three things:

* We have a find by username method on User, use that instead
* You don't catch any errors, and our catch-all express middleware won't catch promise rejections
* You don't account for the fact user could be null